### ISSUE TITLE
Explaination of variable dimensional input for tf.keras.layers.MultiHeadAttention

### ISSUE THREAD

**Issue by girishvn** [keras-team/keras#19241]
Hi, 

I have seen a few examples (such as `https://keras.io/examples/vision/mobilevit/`) where an input of shape `B x T x W x dim` is successfully pushed through a `MultiHeadAttention` layer. From the documentation it seems like only a shape of `B x T x dim` should be accepted, but on further investigation of the source code it seems like variable shaped tensors are accepted with use of some `einsum` dense logic. 

I've had difficult tracking exactly what happens so an explanation would be helpful. 
Is it such that an input of shape `B x T x W x dim` is collapsed into a tensor of shape `B x T x W *dim `?

Thanks.

**Comment by nkovela1** [comment]
Hi @girishvn, would you be interesting in helping with documentation here? Feel free to open a PR and tag us for review. Thanks!
